resources/views/date/rehearsal/listAllAttendances.blade.php

```diff
@@ -0,0 +1,151 @@
+                                                                $iconclass = "fa-minus";
+                                                                break;
+                                                        }?>
+                                                        <td class="{{$tdclass}}">
+                                                            <i class="fa {{$iconclass}}"></i>
+                                                            @if($rehearsal->hasCommented($user))
+                                                                <?php $comment = $rehearsal->getComment($user);?>
+                                                                &nbsp;
+                                                                <a class="btn btn-2d btn-toggle comment-toggle">
+                                                                    <i class="far fa-comment" title="{{$comment}}"></i>
+                                                                </a>
+                                                                <div class="full-comment" style="display: none"> {{$comment}} </div>
+                                                            @endif
+                                                        </td>
+                                                    @endforeach
+                                                </tr>
+                                            @endforeach
+                                        @endforeach
+                                    @endforeach
+                                </tbody>
+                            </table>
+                        </div>
+                        @if(!is_array($rehearsals))
+                            {{ $rehearsals->links() }}
+                        @endif
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+@endsection
+
+@section('js')
+    <script type="text/javascript">
+        $(document).ready(function () {
+            // open and close subvoices
+            $(".subvoice").click(function (){
+                var togglebutton = $(this).find(".btn-toggle");
+                var voice = togglebutton.data('voice');
+
+                $(this).nextAll(".voice-"+voice).toggle();
+                if("hidden" === togglebutton.data("status")){
+                    togglebutton.data("status","display").children("i").removeClass("fa-caret-right").addClass("fa-caret-down");
+                }
+                else{
+                    togglebutton.data("status", "hidden").children("i").removeClass("fa-caret-down").addClass("fa-caret-right");
+                }
+            });
+            // open and close comments for one user
+            $("td").has(".comment-toggle").click(function(){
+                $(this).siblings().addBack().find(".full-comment").toggle();
+            });
+        });
+    </script>
+@endsection
```

Comment:
Accessing this page generates more than 400 SQL queries, almost all of which are duplicate user queries. This slows down the page considerably. The number of duplicate queries should be 0 and the number of overall queries should be no more than 10. Of course, both of those numbers are soft limits and have some leeway, especially if there is a good reason or if the amount of work required to reduce the number is unreasonable. But 400 is too much.

The gig equivalent probably has the same issue.